src/traces/splom/attributes.js

```diff
@@ -0,0 +1,121 @@
+/**
+* Copyright 2012-2018, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var scatterGlAttrs = require('../scattergl/attributes');
+var cartesianIdRegex = require('../../plots/cartesian/constants').idRegex;
+
+function makeAxesValObject(axLetter) {
+    return {
+        valType: 'info_array',
+        freeLength: true,
+        role: 'info',
+        editType: 'calc',
+        items: {
+            valType: 'enumerated',
+            values: [cartesianIdRegex[axLetter].toString(), ''],
+            editType: 'plot'
+        },
+        description: [
+            '..'
+        ].join(' ')
+    };
+}
+
+module.exports = {
+    dimensions: {
+        _isLinkedToArray: 'dimension',
+
+        visible: {
+            valType: 'boolean',
+            role: 'info',
+            dflt: true,
+            editType: 'calc',
+            description: ''
+        },
+        label: {
+            valType: 'string',
+            role: 'info',
+            editType: 'calc',
+            description: ''
+        },
+        values: {
+            valType: 'data_array',
+            role: 'info',
+            editType: 'calc+clearAxisTypes',
+            description: [
+                ''
+            ].join(' ')
+        },
+
+        // TODO should add an attribute to pin down x only vars and y only vars
+        // like https://seaborn.pydata.org/generated/seaborn.pairplot.html
+        // x_vars and y_vars
+
+        editType: 'calc+clearAxisTypes',
+        description: [
+            '..'
+        ].join(' ')
+    },
+
+    mode: scatterGlAttrs.mode,
+    text: scatterGlAttrs.text,
+
+    marker: scatterGlAttrs.marker,
+
+    line: scatterGlAttrs.line,
+    connectgaps: scatterGlAttrs.connectgaps,
+
+    xdirection: {
+        valType: 'enumerated',
+        values: ['right', 'left'],
+        dflt: 'right',
+        role: 'info',
+        editType: 'plot',
+        description: ''
+    },
+    ydirection: {
+        valType: 'enumerated',
+        values: ['top', 'bottom'],
+        dflt: 'bottom',
+        role: 'info',
+        editType: 'plot',
+        description: ''
+    },
```

Comment:
`layout.grid` doesn't have an analog of `xdirection`, do we need that? And in place of `ydirection` we have `roworder: ('top to bottom'|'bottom to top')` which feels clearer to me than `('top'|'bottom')`. Actually I wonder if these belong in `splom` attributes at all, or if we should just let `layout.grid` handle it. It's nice for the simple single-trace case, but once you get two `splom` traces, seems like only one (the first?) matters, so having one in each trace would be confusing.

Reply:
> Actually I wonder if these belong in splom attributes at all, or if we should just let layout.grid handle it. It's nice for the simple single-trace case, but once you get two splom traces, seems like only one (the first?) matters, so having one in each trace would be confusing.

I agree 100% here. I'm thinking of :hocho: ing `xdirection` and `ydirection`. Users won't have a way to plot their splom dimensions from right to left until `layout.grid.columnorder` is implemented, but I don't think that's a big deal. Any objections?